### PR TITLE
Compute logits*V in FP32 in attention kernel to improve numerical accuracy

### PR DESCRIPTION
**Pull request by beginlner**
Before this PR: Cast `logits` from FP32 to BF16/FP16, then compute `dot(logits, V)`.
After this PR: Cast `V` from BF16/FP16 to FP32, then compute `dot(logits, V)`.

### CODE REVIEW

**Comment by WoosukKwon** [comment]
Hi @beginlner, apologies for the late review. Could you elaborate more on this PR? For example, why do you think we need this change? Do you see any accuracy issue due to this data type choice?

**Comment by mgoin** [comment]
Closing for lack of activity and justification
